src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/SyntaxTokenToSemanticTokensEditHelper.cs

```diff
@@ -16,84 +16,109 @@ internal static class SyntaxTokenToSemanticTokensEditHelper
-                    }
-                };
-                return result;
+                        else
+                        {
+                            results.Add(new SemanticTokensEdit
+                            {
+                                Start = diff.Position,
+                                Data = new uint[] { newArray[diff.NewTextPosition.Value] },
+                                DeleteCount = 0,
+                            });
+                        }
+                        break;
+                }
             }
 
-            if (startIndex < dataLength)
+            return results;
+        }
+    }
+
+    internal class UIntArrayDiffer : TextDiffer
+    {
+        private UIntArrayDiffer(uint[] oldArray, uint[] newArray)
+        {
+            OldArray = oldArray;
+            NewArray = newArray;
+        }
+
+        private uint[] OldArray { get; }
+        private uint[] NewArray { get; }
+
+        protected override int OldTextLength => OldArray.Count();
+        protected override int NewTextLength => NewArray.Count();
+
+        protected override bool ContentEquals(int oldTextIndex, int newTextIndex)
+        {
+            return OldArray[oldTextIndex] == NewArray[newTextIndex];
+        }
+
+        public static IReadOnlyList<DiffEdit> GetEdits(uint[] oldArray, uint[] newArray)
+        {
+            if (oldArray is null)
             {
-                return new SemanticTokensEditCollection
-                {
-                    ResultId = newTokens.ResultId,
-                    Edits = new[] {
-                        new SemanticTokensEdit {
-                            Start = startIndex,
-                            DeleteCount = 0,
-                            Data = ImmutableArray.Create(newTokens.Data, startIndex, newTokens.Data.Length - startIndex)
-                        }
-                    }
-                };
+                throw new ArgumentNullException();
             }
-
-            if (startIndex < prevDataLength)
+            if (newArray is null)
             {
-                return new SemanticTokensEditCollection
-                {
-                    ResultId = newTokens.ResultId,
-                    Edits = new[] {
-                        new SemanticTokensEdit {
-                            Start = startIndex,
-                            DeleteCount = prevDataLength - startIndex
-                        }
-                    }
-                };
+                throw new ArgumentNullException();
             }
 
-            return new SemanticTokensEditCollection
+            if (oldArray.SequenceEqual(newArray))
             {
-                ResultId = newTokens.ResultId,
-                Edits = Array.Empty<SemanticTokensEdit>()
-            };
+                return Array.Empty<DiffEdit>();
+            }
+            else if (oldArray.Length == 0 || newArray.Length == 0)
+            {
+                throw new NotImplementedException();
```

Comment:
I'm hitting this exception if I use multiple carets and _ENTER_ to split multiple `<SurveyPrompt />` tags.